Source/Core/Core/PowerPC/JitArm64/JitAsm.cpp

```diff
@@ -100,15 +100,12 @@ void JitArm64::GenerateAsm()
   if (assembly_dispatcher)
   {
     // set the mem_base based on MSR flags
-    LDR(IndexType::Unsigned, ARM64Reg::W28, PPC_REG, PPCSTATE_OFF(msr));
-    FixupBranch physmem = TBNZ(ARM64Reg::W28, 31 - 27);
-    MOVP2R(MEM_REG,
+    LDR(IndexType::Unsigned, ARM64Reg::W27, PPC_REG, PPCSTATE_OFF(msr));
+    TST(ARM64Reg::W27, LogicalImm(1 << (31 - 27), 32));
+    MOVP2R(MEM_REG, jo.fastmem_arena ? memory.GetLogicalBase() : memory.GetLogicalPageMappingsBase());
+    MOVP2R(ARM64Reg::X25,
```

Comment:
Please put the TST after the MOVP2Rs (both here and in the other instances of code like this). Putting the TST right after the LDR means that in-order CPUs won't be able to run the MOVP2Rs until the load completes.